lib/platform.h

```diff
@@ -144,21 +146,19 @@ namespace cppcheck {
                 return "unix64";
             case PlatformFile:
                 return "platformFile";
-            default:
```

Comment:
I like that you remove this default case. But well now the function seems to be more fragile.

I suggest removing the condition before the switch, and put a unconditional `return "unknown" below the switch.

Reply:
Just above the switch statement I have setup a conditional to handle values outside the range of the enum.

Reply:
Yes that seems fragile. If I add an enum type there is a risk that the function will not return any string. If the enum constants are reordered the conditional might not work properly.

Reply:
I do want to keep the Wswitch warnings. but people sometimes don't care about those. There was recently a Wswitch warning that was active for several days.

Reply:
My "preferred" resolution, and what I've done in other code, is to define MIN_[name] and MAX_[name] in the enum declaration. Set the lowest constant to MIN_[name] and set MAX_[name] to the last value. Any 'if' condition tests would be against MIN/MAX values. That seemed like overkill in this situation.